packages/core/src/SceneManager.ts

```diff
@@ -28,27 +28,23 @@ export class SceneManager {
   /**
    * Add scene.
    * @param scene - The scene which want to be added
+   * @param scene - The scene which want to be added
    */
-  addScene(scene: Scene): void;
-
-  /**
-   * Add scene at specified index.
-   * @param index - specified index
-   * @param child - The scene which want to be added
-   */
-  addScene(index: number, scene: Scene): void;
-
-  addScene(indexOrScene: number | Scene, scene?: Scene): void {
-    if (typeof indexOrScene === "number") {
-      this._scenes.add(indexOrScene, scene);
-    } else {
-      scene = indexOrScene;
-      this._scenes.push(scene);
-    }
-
+  addScene(scene: Scene, index?: number): void {
```

Comment:
Function overloading is the advantage of TS. Different parameters can use independent function signatures and annotations. If it is JS, I will prefer your modification method, and TS will prefer the current method.